src/game/logic/ai/turretai.cpp

```diff
@@ -13,3 +13,651 @@
  *            LICENSE
  */
 #include "turretai.h"
+#include "aiupdate.h"
+#include "gamelogic.h"
+#include "partitionmanager.h"
+#include "randomvalue.h"
+#include "terrainlogic.h"
+
+StateReturnType Frame_To_Sleep_Time(unsigned int i1, unsigned int i2, unsigned int i3, unsigned int i4)
+{
+    if (i1 > i2) {
+        i1 = i2;
+    }
+
+    if (i1 > i3) {
+        i1 = i3;
+    }
+
+    if (i1 > i4) {
+        i1 = i4;
+    }
+
+    unsigned int frame = g_theGameLogic->Get_Frame();
+
+    if (i1 <= frame) {
```

Comment:
flip, and args are same as updatemodule Frame_To_Sleep_Time

Reply:
Fixed.